yew-macro/src/html_tree/html_tag/mod.rs

```diff
@@ -137,95 +133,205 @@ impl ToTokens for HtmlTag {
+        } else {
+            let attrs = attributes.iter().map(
+                |TagAttribute {
+                     label,
+                     question_mark,
+                     value,
+                 }| {
+                    let key = label.to_lit_str();
+                    if question_mark.is_some() {
+                        let sr = stringify::stringify_option_at_runtime(value);
+                        quote! {
+                            ::yew::virtual_dom::PositionalAttr(#key, #sr)
+                        }
+                    } else {
+                        let sr = value.stringify();
+                        quote! {
+                            ::yew::virtual_dom::PositionalAttr::new(#key, #sr)
+                        }
                     }
-                    None
-                }
-                None => Some(quote! {
-                    let __yew_classes
-                        = ::std::convert::Into::<::yew::virtual_dom::Classes>::into(#classes);
+                },
+            );
+            Some(quote! {
+                #vtag.attributes = ::yew::virtual_dom::Attributes::Vec(::std::vec![#(#attrs),*]);
+            })
+        };
+
+        let push_booleans = if booleans.is_empty() {
+            None
+        } else {
+            let tokens = booleans
+                .iter()
+                .map(|TagAttribute { label, value, .. }| {
+                    let label_str = label.to_lit_str();
+                    let sr = label.stringify();
+                    quote_spanned! {value.span()=> {
+                        if #value {
+                            #vtag.__macro_push_attribute(#label_str, #sr);
+                        } else {
+                            #vtag.__macro_push_attribute_placeholder(#label_str);
+                        };
+                    }}
+                })
+                .collect::<TokenStream>();
+            Some(tokens)
+        };
+
+        let push_classes = match classes {
+            Some(ClassesForm::Tuple(classes)) => {
+                let sr = stringify::stringify_at_runtime(quote! { __yew_classes });
+                Some(quote! {
+                    let __yew_classes = ::yew::virtual_dom::Classes::default()
+                        #(.extend(#classes))*;
+
                     if !__yew_classes.is_empty() {
-                        #vtag.push_attribute(
-                            "class",
-                            ::std::string::ToString::to_string(&__yew_classes),
-                        );
+                        #vtag.__macro_push_attribute("class", #sr);
+                    } else {
+                        #vtag.__macro_push_attribute_placeholder("class");
                     };
-                }),
+                })
+            }
+            Some(ClassesForm::Single(classes)) => match classes.try_into_lit() {
+                Some(lit) => {
+                    if lit.value().is_empty() {
+                        None
```

Comment:
Should we push a placeholder attribute here?

Reply:
The reason I didn't is because this is the literal case (i.e. `html! { <div class="weird-button"></div> }`) so the value never changes. The only reason for the `is_empty()` check is so that `html! { <div class=""></div> }` doesn't add the class attribute (though I have to wonder whether we shouldn't have that be a warning because I don't see this ever being useful).

Reply:
Ah yeah great point